sklearn/base.py

```diff
@@ -768,7 +804,8 @@ def is_classifier(estimator):
     out : bool
         True if estimator is a classifier and False otherwise.
     """
-    return getattr(estimator, "_estimator_type", None) == "classifier"
+    estimator_tags = getattr(estimator, "_get_tags", dict)()
+    return estimator_tags.get("estimator_type", None) == "classifier"
```

Comment:
The tag should have a default in `_DEFAULT_TAGS` instead of relying on that kind of logic

Reply:
I was having some issues due to #17836. Now that #17837 has been merged, I will simplify this.